### ISSUE TITLE
deploy doesn't run in build stage

### ISSUE THREAD

**Issue by cjw296** [travis-ci/travis-ci#8337]
Here's the .travis.yaml:

https://github.com/Simplistix/testfixtures/blob/e533404075c6d369f1353df3efef4866b7cadc51/.travis.yml

Here's the build log:

https://travis-ci.org/Simplistix/testfixtures/jobs/271477850

Even removing the `on` condition has had no effect.

**Comment by cjw296** [comment]
Okay, found the bug, here's the workaround:

https://github.com/Simplistix/testfixtures/commit/990b39c46f36d83fdaf4e7bea2fea1547d109691

If you `after_success: skip`, your `deploy` section never gets run. However, I *want* to skip `after_success` here, since I can't (and don't want to!) report coverage information from my release stage:

https://travis-ci.org/Simplistix/testfixtures/jobs/271481969#L174

**Comment by BanzaiMan** [comment]
Thanks for opening the issue. This is not obvious, but the deployment kicks in as the step after `after_success`. As you observed, if you skip `after_success` with `skip`, the deployment is skipped, too.

This is probably not the best thing to do.

For now, you can execute the `after_success` step with an explicit `true` or `echo "skipped"` or something similar.

**Comment by cjw296** [comment]
Indeed, doesn't make sense to me, but at least document it on https://docs.travis-ci.com/user/customizing-the-build#The-Build-Lifecycle please :-)

**Comment by webknjaz** [comment]
I'd add that it worked well earlier :) At least, it seemed to work
